### ISSUE TITLE
部署成功了，但是识别图片结果为什么看起来很离谱

### ISSUE THREAD

**Issue by kimiller** [Vision-CAIR/MiniGPT-4#146]
![image](https://user-images.githubusercontent.com/102167137/234212771-3d8fa7da-b570-4a5c-81ff-129ff18dc69b.png)

识别结果为什么这么离谱呢，是一个小白按着教程部署的，不知道哪里有问题

**Comment by SnakeHacker** [comment]
<img width="1333" alt="image" src="https://user-images.githubusercontent.com/11294011/234239597-56e84f1d-c959-43ba-9509-691d75379589.png">

**Comment by silent780** [comment]
Same issue,
Web demo did a good job, but my local model is just look like a shit, my card is 4090 and model is 13B. Is this just normal or something wrong with my project?

**Comment by Korner83** [comment]
You might loaded in the wrong weights, I would advise to double check the settings in the yml files to the proper models. Have you followed the steps in the description?

**Comment by kimiller** [comment]
I didn't follow steps as  in the description.
I have used below cmd to deploy it into Google Colab, actually I don't know what happened in the cmd,I think I need redeploy it and use another method 
cmd like this ：

!git clone -b dev https://github.com/camenduru/minigpt4
!wget https://huggingface.co/ckpt/minigpt4/resolve/main/minigpt4.pth -O /content/minigpt4/checkpoint.pth
!wget https://huggingface.co/ckpt/minigpt4/resolve/main/blip2_pretrained_flant5xxl.pth -O /content/minigpt4/blip2_pretrained_flant5xxl.pth

!pip install -q salesforce-lavis
!pip install -q bitsandbytes
!pip install -q accelerate
!pip install -q gradio==3.27.0
!pip install -q git+https://github.com/huggingface/transformers.git -U

%cd /content/minigpt4
!python app.py

**Comment by kimiller** [comment]
@SnakeHacker 哈哈哈笑死我了

**Comment by kimiller** [comment]
@Korner83  Thank you!!!Let me try it

**Comment by youyuanrsq** [comment]
> 

6

**Comment by SnakeHacker** [comment]
<img width="1397" alt="image" src="https://user-images.githubusercontent.com/11294011/234501786-8e8058cc-9e97-4487-821b-6276ccf10aa5.png">
模型骂人了，学到了国粹。。

**Comment by thiner** [comment]
我的demo第一次也出现了这种胡说八道的情况。后来发现是我在修改`minigpt4/configs/models/minigpt4.yaml`文件时，误将`llama_model`指向了vicuna的delta文件。你不一定是跟我一样的情况，但是建议仔细检查一下配置文件是否正确。

**Comment by kimiller** [comment]
@thiner 
![image](https://user-images.githubusercontent.com/102167137/234533887-d899695d-aa5a-4bdb-8337-191004bb2173.png)
我测了一下minigpt4 在线demo的情况，也是有乱说的时候，识图的功能应该还没这么完善吧，不过这已经做的不错了

**Comment by chenchuntan** [comment]
有没有最低要求的配置  我电脑不行 CPU是I5 显卡是1060的

**Comment by youyuanrsq** [comment]
> 有没有最低要求的配置 我电脑不行 CPU是I5 显卡是1060的

跑7B的模型最低要12G的显存

**Comment by YiyangZhou** [comment]
I have tried many models and found that mPLUG-Owl https://github.com/X-PLUG/mPLUG-Owl seems to have much stronger visual capabilities than the others, with more accurate image recognition

**Comment by vateye** [comment]
> @YiyangZhou thank you, tested this one, but it seems to be a copy of MiniGPT-4.

No, I have tested it. It seems to perform better than miniGPT4 with the smaller model (ViT-L and LLaMA 7B only).

**Comment by kimiller** [comment]
> > @YiyangZhou thank you, tested this one, but it seems to be a copy of MiniGPT-4.
> 
> No, I have tested it. It seems to perform better than miniGPT4 with the smaller model (ViT-L and LLaMA 7B only).

@vateye  yes.I agree with that  it's performance better than miniGPT4 ,  it looks like just made some optimizations, no essential changes.

**Comment by kimiller** [comment]
> > @YiyangZhou thank you, tested this one, but it seems to be a copy of MiniGPT-4.
> 
> No, I have tested it. It seems to perform better than miniGPT4 with the smaller model (ViT-L and LLaMA 7B only).

@vateye  yes.I agree with that it's performance better than miniGPT4 . looks like just made some optimizations , no essential changes.

**Comment by YuzhouPeng** [comment]
> https://github.com/X-PLUG/mPLUG-Owl

Could you please tell me what models have you tried? I may want to try these models also. Thank you so much!

**Comment by YuzhouPeng** [comment]
> I have tried many models and found that mPLUG-Owl https://github.com/X-PLUG/mPLUG-Owl seems to have much stronger visual capabilities than the others, with more accurate image recognition
Could you please tell me what models have you tried? I may want to try these models also. Thank you so much!

**Comment by YuzhouPeng** [comment]
> I have tried many models and found that mPLUG-Owl https://github.com/X-PLUG/mPLUG-Owl seems to have much stronger visual capabilities than the others, with more accurate image recognition

Could you please tell me what models have you tried? I may want to try these models also. Thank you so much!

**Comment by YiyangZhou** [comment]
> > I have tried many models and found that mPLUG-Owl https://github.com/X-PLUG/mPLUG-Owl seems to have much stronger visual capabilities than the others, with more accurate image recognition
> 
> Could you please tell me what models have you tried? I may want to try these models also. Thank you so much!

Minigpt4:

**Comment by YiyangZhou** [comment]
> > I have tried many models and found that mPLUG-Owl https://github.com/X-PLUG/mPLUG-Owl seems to have much stronger visual capabilities than the others, with more accurate image recognition
> 
> Could you please tell me what models have you tried? I may want to try these models also. Thank you so much!

minigpt4:https://github.com/Vision-CAIR/MiniGPT-4
llava:https://github.com/haotian-liu/LLaVA
open_flamingo:https://github.com/mlfoundations/open_flamingo
mPLUG-Owl:https://github.com/X-PLUG/mPLUG-Owl 
BLIP2:
(1)Try it with the transformers package：https://huggingface.co/docs/transformers/index
(2)lavis:https://github.com/salesforce/LAVIS

**Comment by sssssshf** [comment]
有没有终端命令的demo阿  必须要用浏览器吗

**Comment by LeoYYG** [comment]
> @thiner ![image](https://user-images.githubusercontent.com/102167137/234533887-d899695d-aa5a-4bdb-8337-191004bb2173.png) 我测了一下minigpt4 在线demo的情况，也是有乱说的时候，识图的功能应该还没这么完善吧，不过这已经做的不错了

部署在 linux 还是 windows 上？我部署ubuntu 上乱码，mathplotlib的问题，字体库都装全了也无果。

**Comment by ry-2023** [comment]
> 

不指向delta，那么是指向hf吗

**Comment by xueweiprince** [comment]
我也试了下在线demo，识别的结果实在是太烂了

**Comment by coding-alt** [comment]
> 图生文可以自己更换实现方案。

**Comment by 631068264** [comment]
> SnakeHacker

你是怎样让他说中文的。。。。

**Comment by zhouyuezhou1** [comment]
@thiner 请问那应该指向哪个文件呢
